All_home_works/hw1/task2.py

```diff
@@ -0,0 +1,30 @@
+"""
+Write down the function, which reads input line-by-line, and find maximum and minimum values.
+Function should return a tuple with the max and min values.
+For example for [1, 2, 3, 4, 5], function should return [1, 5]
+We guarantee, that file exists and contains line-delimited integers.
+To read file line-by-line you can use this snippet:
+with open("some_file.txt") as fi:
+    for line in fi:
+        ...
+"""
+from typing import Tuple
+
+
+def find_maximum_and_minimum(file_name: str) -> Tuple[int, int]:
+    with open(file_name) as data_file:
+        min_value = max_value = int(data_file.readline()[:-1])
+        for line in data_file.readlines():
+            line = line[:-1]
```

Comment:
This operation does not show your intention.

- please add a test for the case when this code will fail.
- please rewrite this line to show your intent clearly.

Reply:
Are you talking about line = line[:-1]?
I just think about this whole part of the code with min and max also readlines().
line = line.rstrip('\n'). Is it ok?

Reply:
> line = line.rstrip('\n'). Is it ok?

yes. 

>  line = line[:-1]

When this code will fail?

Reply:
I thought about the length of the line. But:
line = ""
line[:-1]
It's ok.
So maybe if a line is None

Reply:
>  So maybe if a line is None

Could deadline() return None?

It is another case.

Reply:
if readline() returns an empty string, the end of the file has been reached.
So I don't know when "str[:-1]" will fail

Reply:
`str[:-1]` will not fail, but it's will have a different result than `line.rstrip('\n')` and your application will get incorrect data.

Reply:
Well. Do you have any more questions?

Reply:
No, only why using the `str[:-1]` in that context is a bug.

Reply:
Sorry, I don't know. The line always ends with '\n' and the last line is empty.

Reply:
Maybe it will be better to use open('text', new line="") and forget about line cutting. But why using the str[:-1] in that context is a bug?

Reply:
>  The line always ends with '\n' and the last line is empty.

So the last line does not come with the `\n`
Can we create a file where the last line is not '\n'?  (Please do experiments)

Reply:
> Maybe it will be better to use open('text', new line="") and forget about line cutting. But why using the str[:-1] in that context is a bug?

Can you give me a working code sample?

Reply:
OK. I understand! If the last line is the last, I mean, no last blank line. My code will fail. Because the line does not end with '\n'.
And about newline="". It's not working. 
Thanks for your patience=)